packages/edit-widgets/src/store/actions.js

```diff
@@ -15,9 +15,11 @@ const WIDGET_AREAS_SAVE_NOTICE_ID = 'WIDGET_AREAS_SAVE_NOTICE_ID';
 /**
  * Yields an action object that setups the widget areas.
  *
+ * @param {Object} savedWidgetAreas Mapping from area ID to array of blocks for overriding published content with drafts and other ongoing changesets.
+ *
  * @yields {Object} Action object.
  */
-export function* setupWidgetAreas() {
+export function* setupWidgetAreas( savedWidgetAreas ) {
```

Comment:
I'm not sure why do we need this change here. Can't we just update the blocks in the store once we edit them? and rely on the fact that the store is always up to date?

Reply:
It's always up to date with the "published" widgets.

This allows us to override specific widget areas with their Changeset counterparts.

Reply:
I'm not certain I grasp the details yet but I think the `edit-widgets` store is used to track the current state of the editor, so maybe we should track the changeset couterpart instead of the published widgets. cc @jorgefilipecosta 

What value do we have in tracking the published widgets there?

Reply:
>...I think the edit-widgets store is used to track the current state of the editor...

Yes

`setupWidgetAreas` loads the published widget areas into the store when the app starts. What I did here was provide a way for the Customizer sync code to replace any combination of those widget areas with what is saved in the Changeset. The published widget areas are mapped into the objects used by the store, but their `blocks` property is set to the Changeset value, if defined:

https://github.com/WordPress/gutenberg/blob/4308984949371bcbabfa68511ad5088e637e5671/packages/edit-widgets/src/store/actions.js#L35

Reply:
It seems we are calling setupWidgetAreas just as a way to replace the blocks with the customizer changeset version. Can't we just do a loop call to updateBlocksInWidgetArea to replace blocks with customizer changeset version, avoiding the need to change this action?

Reply:
See https://github.com/WordPress/gutenberg/pull/16204#discussion_r299709914.